apps/example-server/src/main/user/user.routes.ts

```diff
@@ -18,3 +26,5 @@ export const userV1Routes = createRouter()
     .openapi(getUserRoute, getUserHandler)
     .openapi(updateUserRoute, updateUserHandler)
     .openapi(deleteUserRoute, deleteUserHandler)
+    .openapi(updateUserProfileRoute, updateUserProfileHandler)
```

Comment:
Correct ordering is important here. You have the update profile picture route - `/user/upload-profile-picture`. Also we have the update user route - `/user/:id`. So there is a chance that upload route will be considered an update route where hono will assume `:id` is `upload-profile-picture`. So, it is always better to put the custom endpoints before the regular CRUD ones.